### PR TITLE
CA-388210: use unique datapaths for concurrent VDI copies

### PR DESCRIPTION
**Pull request by edwintorok**
Testing on a box shows that the new unique arguments for 'domain' work, but opening as a draft PR, because this needs wider testing in XenRT.

### CODE REVIEW

File: ocaml/xapi-storage-script/main.ml [hunk 1]
```diff
@@ -49,6 +49,23 @@ let backend_backtrace_error name args backtrace =
...
   backend_error "MISSING_URI" ["Please include a URI in the device-config"]
 
+(** return a unique 'domain' string for Dom0, so that we can plug disks
+  multiple times (e.g. for copy).
+
+  XAPI should give us a unique 'dp' (datapath) string, e.g. a UUID for storage migration,
+  or vbd/domid/device.
+  For regular guests keep the domain as passed by XAPI (an integer).
+ *)
+let domain_of ~dp ~vm' =
```
Comment (lindig) [inline ocaml/xapi-storage-script/main.ml:59]: Requiring a prime in named label is a bit awkward - is there a reason this is not just `~vm`?

**Review by lindig — APPROVED** [review]

File: ocaml/xapi-storage-script/main.ml (hunk 1 above)
Comment (edwintorok) [reply ocaml/xapi-storage-script/main.ml]: All the parameters where this function is called were called `vm'`, however renaming should be easy, I've added a commit at the beginning of this PR that renames `vm'` to `vm`, and then we can avoid having `'` in the labeled argument.

**Comment by edwintorok** [comment]
I rebased and retargeted this to feature/perf and removed the commit that enables concurency (since that one still triggers a deadlock that we're trying to track down in the storage code).

**Review by psafont — APPROVED** [review]

**Comment by lindig** [comment]
Can we merge this?

**Pull request merged**